### DISCUSSION TITLE
var_dump equivalent

### DISCUSSION THREAD

**Discussion by tthiery** [dotnet/runtime#61251] [Q&A]
In PHP there is a function var_dump which dumps a object tree as a string.

I have recently experienced that Blazor (maybe razor) is doing something like that when emitting an object. Is that available in the BCL or just in razor?

**Comment by huoyaoyuan** [comment]
No, and should never be.

Dumping *arbitrary* fields of objects can lead to severe security risk. When running in debugger, you have full control to the target process, so you can inspect any field you want.

> I have recently experienced that Blazor (maybe razor) is doing something like that when emitting an object.

It should be actually JSON serializing. JSON serialization can be easily applied on simple types, but definitely on arbitrary ones.

**Reply by tthiery** [reply]
Well.. in that perspective every Console.WriteLine or any serializer is a security risk.

We do a lot for the newbies right now: top level statements, implicit usings or file scoped namespaces. In addition to .NET Interactive and free available tools. Why not giving them an quick dump tool?

PS: that Razor is JSON serializing makes definitely sense. That is already super close to var_dump but not the same I think.

**Reply by huoyaoyuan** [reply]
> Why not giving them an quick dump tool?

Records can be easily dumped. Objects with complex tree can be hardly dumped though.
